packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,87 @@
-import * as React from "react";
-import { CustomCell, CustomRenderer, drawTextCell, GridCellKind } from "@glideapps/glide-data-grid";
+import React from "react";
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly format: "date" | "time" | "datetime-local";
 }
 
+export const formatValueForHTMLInput = (format: "date" | "time" | "datetime-local", date: Date): string => {
+    if (format === "date") {
+        return date.toISOString().split("T")[0];
+    }
+    if (format === "time") {
+        return date.toISOString().split("T")[1].replace("Z", "");
+    }
+    if (format === "datetime-local") {
+        return date.toISOString().replace("Z", "");
+    }
+    return "";
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, format } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
+    const value = formatValueForHTMLInput(format, cellDataDate);
+
+    return (
+        <input
+            required
+            style={{ minHeight: 26, border: "none", outline: "none" }}
+            type={format}
+            value={value}
+            autoFocus={true}
+            onChange={event => {
+                // handle when clear is clicked and value has been wiped
+                if (event.target.value === "") {
+                    try {
+                        cell.onChange({
```

Comment:
What are the circumstances you expect for this try to throw?

Reply:
in the case the displayDate is some weird string somehow.

Reply:
Can you point me to the code you think would throw? Perhaps the smell here is that we should be protecting at the far end.

Reply:
👍